Readonly keys (#1400)
cap-js/docs: guides/providing-services.md

By default [`virtual` elements](../cds/cdl#virtual-elements) are also _calculated_.
::: tip
The same applies for fields with the [OData Annotations](../advanced/odata#annotations) `@FieldControl.ReadOnly` (static), `@Core.Computed`, or `@Core.Immutable` (the latter only on UPDATEs).
:::

> ::: warning

Reviewer: ```suggestion
::: warning Not allowed on keys
```

> Read-only keys in all variants are not supported and the respective annotation is ignored.

Reviewer: ```suggestion Not allowed on keys
Do not use the `@readonly` annotation on keys in all variants.
```

> :::

Reviewer: The information in the previous tip is similar, right? Should we remove the tip and just keep the warning stating that everything that is read-only will be ignored? A tip for annotations and a warning for keys directly following each other can make a wrong impression.

Reviewer: Ping @vkozyura @David-Kunz

Reviewer: Reading it again three months later, I think I misunderstood it. Here's my current understanding:
This sentence is meant to say "you can't annotate keys with @readonly." So, whereas all the rest of this section talks about the effect of readonly, this sentence has a different angle.

Is my current understanding correct?

Reviewer: @vkozyura Is my current understanding (see my last comment) correct?
